### ISSUE TITLE
test_pages_huge:test/unit/pages.c fails on ARM (armv7l)

### ISSUE THREAD

**Issue by marxin** [jemalloc/jemalloc#1045]
Testing version 5.0.1:

```
./configure --host=armv7hl-suse-linux-gnueabi --build=armv7hl-suse-linux-gnueabi --program-prefix= --disable-dependency-tracking --prefix=/usr --exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/lib --libexecdir=/usr/lib --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --disable-dependency-tracking --enable-cc-silence --enable-prof
```

I spot:

```
[  266s] === test/unit/pages ===
[  266s] test_pages_huge:test/unit/pages.c:15: Failed assertion: (pages_huge(hugepage, HUGEPAGE)) != (config_thp) --> true == true: Unexpected pages_huge() result
[  266s] test_pages_huge:test/unit/pages.c:17: Failed assertion: (jet_pages_nohuge(hugepage, ((size_t)(1U << 21)))) == (false) --> true != false: Unexpected pages_nohuge() result
[  266s] test_pages_huge (non-reentrant): fail
[  266s] test_pages_huge:test/unit/pages.c:15: Failed assertion: (pages_huge(hugepage, HUGEPAGE)) != (config_thp) --> true == true: Unexpected pages_huge() result
[  266s] test_pages_huge:test/unit/pages.c:17: Failed assertion: (jet_pages_nohuge(hugepage, ((size_t)(1U << 21)))) == (false) --> true != false: Unexpected pages_nohuge() result
[  266s] test_pages_huge (libc-reentrant): fail
[  266s] test_pages_huge:test/unit/pages.c:15: Failed assertion: (pages_huge(hugepage, HUGEPAGE)) != (config_thp) --> true == true: Unexpected pages_huge() result
[  266s] test_pages_huge:test/unit/pages.c:17: Failed assertion: (jet_pages_nohuge(hugepage, ((size_t)(1U << 21)))) == (false) --> true != false: Unexpected pages_nohuge() result
[  266s] test_pages_huge (arena_new-reentrant): fail
[  266s] --- pass: 0/3, skip: 0/3, fail: 3/3 ---
```

**Comment by lheckemann** [comment]
Yep, I'm getting this too. I think it's a configuration issue, i.e. `JEMALLOC_HAVE_MADVISE_HUGE` as used in [the implementation](https://github.com/jemalloc/jemalloc/blob/9f455e2786685b443201c33119765c8093461174/src/pages.c#L315) should be defined as false on ARMv7.

**Comment by davidtgoldblatt** [comment]
Fixed by #1102. Thanks!
